firefox-ios/Client/Experiments/Messaging/GleanPlumbMessageStore.swift

```diff
@@ -60,11 +60,12 @@ class GleanPlumbMessageStore: GleanPlumbMessageStoreProtocol {
         set(key: message.id, metadata: message.metadata)
     }
 
-    /// For the MVP, we always expire the message.
-    func onMessagePressed(_ message: GleanPlumbMessage) {
-        onMessageExpired(message.metadata,
-                         surface: message.surface,
-                         shouldReport: false)
+    func onMessagePressed(_ message: GleanPlumbMessage, shouldExpire: Bool) {
+        if shouldExpire {
```

Comment:
Instead of checking if a message is a microsurvey and not let it expire, I decide to add a generic boolean in which any message can determine whether it should expire or not. This seems to be more flexible for the infrastructure in my opinion.

We may diverge from Android on this implementation detail. Let me know if there are any concerns or if we should be checking specifically if its a microsurvey.